### PR TITLE
Add document view only mode

### PR DESCRIPTION
**Pull request by cs2be**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,24 @@ We appreciate contribution to various aspects of the platform and supporting con
...
+## PaddlePaddle.org Document Generation and Viewer Tool
 
 **Prerequisites**:  Docker is required in order to run PaddlePaddle.org doucmentation viewer.
 [Install Docker Here](https://docs.docker.com/engine/installation/).
 
+PaddlePaddle.org not only powers [http://paddlepaddle.org](http://paddlepaddle.org), it can be used as a tool for documentation creators to generate and view their documentation on their local machine.  To run PaddlePaddle.org as a Document tool, please follow the instructions below:
+
+### Documentation Viewer Only Tool
+
+####1) Run PaddlePaddle.org Docker Image.
```
Comment (varunarora) [inline README.md]: I think you need to space after the last # for it to format with Github. Your editor is slightly lax on that rule.

File: portal/portal/settings.py [hunk 2]
```diff
@@ -25,19 +25,43 @@
...
+    PPO has 3 modes:
+    1)  Default:  Default website mode.
+    2)  DOC_EDIT_MODE: Document editor mode.  This will allow document editors to generate and view
+        their documentation.  This mode is activated if there is no 'ENV' environment variable set and
+        HAS_MOUNT is NOT set or set to '1' (ie:  We have mounted a volume to /var/content in Docker).
+    3)  DOC_VIEW_MODE: Document viewer mode.  This will allow users to view the latest PaddlePaddle
+        documentation.  This mode is activated if there is no 'ENV' environment variable set AND
+        'HAS_MOUNT' is set to 0 (meaning there is no mount set for the content directory)
+    '''
+    Default, DOC_EDIT_MODE, DOC_VIEW_MODE = range(3)
```
Comment (varunarora) [inline portal/portal/settings.py]: That's cool!!

File: portal/portal/settings.py [hunk 3]
```diff
@@ -25,19 +25,43 @@
...
+    else:
+        CURRENT_PPO_MODE = PPO_MODES.DOC_EDIT_MODE
+
     DEBUG = True
+
 elif ENV == 'development':
     DEBUG = True
 
-DEFAULT_DOCS_VERSION = 'develop' if not DOC_MODE else 'doc_test'
+DEFAULT_DOCS_VERSION = 'develop' if CURRENT_PPO_MODE != PPO_MODES.DOC_EDIT_MODE else 'doc_test'
```
Comment (varunarora) [inline portal/portal/settings.py]: Even doc preview mode should assume develop right?

File: portal/portal/portal_helper.py [hunk 4]
```diff
@@ -81,4 +85,29 @@ def folder_name_for_content_id(content_id):
...
+    return os.path.isfile(dest_file_path)
+
+
+def download_and_extract_workspace():
+    dest_file_path = '%s/%s' % (settings.CONTENT_DIR, settings.WORKSPACE_ZIP_FILE_NAME)
+
+    if not os.path.isfile(dest_file_path):
+        r = requests.get(settings.WORKSPACE_DOWNLOAD_URL, stream=True)
+        with open(dest_file_path, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=1024):
```
Comment (varunarora) [inline portal/portal/portal_helper.py]: I think you don't actually need to do the iter_content stuff - but if you did it intentionally or if it just works fine, awesome :)

Comment (varunarora) [reply portal/portal/portal_helper.py]: Oh wait you are `stream=True`ing

File: portal/portal/settings.py [hunk 5]
```diff
@@ -100,7 +124,7 @@
 CACHES = {
     'default': {
         'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
-        'TIMEOUT': 5 if DEBUG else 300
+        'TIMEOUT': 0 if DEBUG else 300
```
Comment (varunarora) [inline portal/portal/settings.py]: 👍

File: portal/portal/sitemap_helper.py [hunk 6]
```diff
@@ -242,12 +242,14 @@ def get_available_versions():
...
     string_based_version = []
     number_based_version = []
-    for version in versions:
-        normalized_version = version.split('.')
-        if len(normalized_version) > 1:
-            number_based_version.append(version)
-        else:
-            string_based_version.append(version)
+
+    if versions:
```
Comment (varunarora) [inline portal/portal/sitemap_helper.py]: 👍

File: portal/portal/views.py [hunk 7]
```diff
@@ -282,17 +283,28 @@ def _get_static_content_from_template(path):
...
+        }
+        return render(request, 'index_doc_mode.html', context)
+
+    elif settings.CURRENT_PPO_MODE == settings.PPO_MODES.DOC_VIEW_MODE:
+        if portal_helper.has_downloaded_workspace_file():
+            preferred_version = portal_helper.get_preferred_version(request)
+            return _redirect_first_link_in_contents(request, preferred_version, 'documentation')
+        else:
+            response = render(request, 'index_doc_view_mode.html')
+            portal_helper.set_preferred_version(request, response, 'develop')
```
Comment (varunarora) [inline portal/portal/views.py]: Okay this resolves my earlier concern about not defaulting to `develop` in DOC_VIEW mode

File: portal/portal/settings.py (hunk 3 above)
Comment (cs2be) [reply portal/portal/settings.py]: yea, only DOC_EDIT_MODE should be using 'doc_test'

**Pull request merged**